src/Implementation/TypeDetector.php

```diff
@@ -148,19 +155,26 @@ private static function determineScalarTypeIfPossible(string $class, ReflectionC
         return null;
     }
 
-    private static function convertClassToType(string $class): Type
+    private static function convertClassToType(ReflectionClass $class): Type
```

Comment:
I don't think you will need to change this: See https://github.com/arnedesmedt/value-objects/pull/1/files#diff-85d8f5891c0f93bf48cba6835bcaec2627f8f3b8e5f89554b50bf00679bdf4f3R139 

It will already return a string Type for your datetime because it implements the string interface. I think you should add an extra if before that line, to check if it implements your datetimevalue interface and then return a JsonSchema::string() with format date-time

Reply:
Yes, that was still something from when I tried to use it without the string implementation.

I've removed it

Reply:
You still need to add a the format to JsonSchema string

Reply:
Right, added!